mocha.el

```diff
@@ -71,11 +71,8 @@ From http://benhollis.net/blog/2015/12/20/nodejs-stack-traces-in-emacs-compilati
 
 (defun mocha-find-project-root ()
   "Find the root of the project."
-  (let ((root-files '("package.json" ".git" ".hg" ".svn")) (dir nil) (i 0))
-    (while (not dir)
-      (setq dir (locate-dominating-file default-directory (nth i root-files)))
-      (setq i (+ i 1)))
-    dir))
+  (let ((root-dir (f--traverse-upwards (f-exists? (f-expand "package.json" it)))))
```

Comment:
It's an anaphoric function, which is a bit magic for sure. 😄 

It works basically like this:

```lisp
(defmacro with-it (&rest body)
  `(let ((it "the thing")) ,@body))

(with-it
 (print it)) ;; "the thing"
```

So the macro will expand into this:

```lisp
(let ((it "the thing")) (print it))
```

It's very handy. Take a look at [dash](https://github.com/magnars/dash.el) where it's used all over the place.